OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java

```diff
@@ -188,10 +189,15 @@ private void addPoint(RoutingContext ctx, Map<String, RegisteredMap> knownMaps,
 		boolean onlyJointMap = true;
 		List<String> regions = new ArrayList<String>();
 		for (BinaryMapDataObject o : resList) {
-			regions.add(or.getDownloadName(o));
-			if (!or.isDownloadOfType(o, OsmandRegions.MAP_JOIN_TYPE)
-					&& !or.isDownloadOfType(o, OsmandRegions.ROADS_JOIN_TYPE)) {
-				onlyJointMap = false;
+			boolean hasMapType = or.isDownloadOfType(o, OsmandRegions.MAP_TYPE);
```

Comment:
not clear what it fixes

Reply:
This is also directly connected to original issue. The problem. We have SRTM-only (sub)regions such as Afghanistan, Egypt, Libya, Chile, etc. That SRTM-only (no map, no routing!!!) regions are added here to regions List and a few lines later it is sorted to get 1st only String from regions.

In case of Afghanistan + Tajikistan, the sorted result is:

afghanistan_baghlan_asia - the longest name, sorted to top = get(0)
afghanistan_asia (ignored)
tajikistan_asia (ignored)

As a result, afghanistan_baghlan_asia is the only mapsToDownload region, which is OBVIOUSLY not a map/roads, not allowed to download as a map/roads and finally we have the EMPTY list of Missing Maps and disabled Download button (as displayed in this issue indeed).

That's it.